lib/index.js

```diff
@@ -28,27 +30,40 @@ module.exports = function( args, options ) {
 	}
 
 	const cwd = getCwd();
-	const mainPackageName = require( path.resolve( cwd, 'package.json' ) ).name;
 	const startTime = process.hrtime();
 	const toolOptions = getOptions( options, cwd );
 	const repositoryResolver = require( toolOptions.resolverPath );
 	const forkPool = createForkPool( CHILD_PROCESS_PATH );
 
+	if ( !options.skipRoot && toolOptions.$rootRepository && !fs.existsSync( path.join( cwd, '.git' ) ) ) {
```

Comment:
For future reference, please extract such complex conditions to a smaller util. It's PITA to understand it without going deeper. 

The goal is to figure out if mrgit should break the process. So, we have a name for the util.

```suggestion
	if ( shouldBreakProcess( toolOptions ) ) {
```

And somewhere below:

```js
function shouldBreakProcess( toolOptions ) {
	if ( options.skipRoot ) {
		return false;
	}

	if ( !toolOptions.$rootRepository ) {
		return false;
	}

	if ( fs.existsSync( path.join( cwd, '.git' ) ) ) {
		return false;
	}

	return true;
}
```

Code style depends on the developer, but all conditions are a bit more readable than it was before.

Reply:
The proposed change: ec53021.

<img width="559" alt="image" src="https://github.com/cksource/mrgit/assets/2270764/b56e0a68-d37a-4850-a522-383d6f8301d3">